tests/zenko_tests/node_tests/iam_policies/cloudserver/IAMUser.js

```diff
@@ -0,0 +1,257 @@
+const assert = require('assert');
+const async = require('async');
+const uuidV4 = require('uuid/v4');
+const {makeGETRequest} = require("../../utils/request");
+
+const iam = require('../../s3SDK').scalityIAMClient;
+const s3 = require('../../s3SDK').scalityS3Client;
+
+const userName = "username-policy-test-metadatasearch2";
+
+const bucketName = "bucket-test-mdsearch1";
+const bucketName2 = "bucket-test-mdsearch2";
+const allowMdSearchPolicyName = "allowMDSearchPolicy";
+const denyMdSearchPolicyName = "denyMDSearchPolicy";
+
+function createPolicy(action, isAllow = true, resource = "*") {
+    return JSON.stringify({
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Sid": "Stmt1644586763301",
+                "Action": [
+                    action,
+                ],
+                "Effect": (isAllow ? "Allow" : "Deny"),
+                "Resource": resource
+            }
+        ]
+    });
+}
+
+function metadataSearchResponseCode(userCredentials, bucketName, cb) {
+    return makeGETRequest(`/${bucketName}/?search=${encodeURIComponent('key LIKE "file"')}`, (err, res) => {
+        if (err)
+            return cb(err);
+        return cb(null, res.statusCode);
+    }, userCredentials)
+}
+
+describe('IAM Policies MetadataSearch', () => {
+
+    let userCredentials = {
+        accessKeyId: "",
+        secretAccessKey: "",
+    };
+
+    before(done => {
+        async.series([
+            next => s3.createBucket({Bucket: bucketName}, next),
+            next => s3.createBucket({Bucket: bucketName2}, next),
+            next => iam.createUser({UserName: userName}, next),
+            next => iam.createAccessKey({UserName: userName}, next)
+        ], (err, res) => {
+            assert.ifError(err);
+            userCredentials.accessKeyId = res[3].AccessKey.AccessKeyId;
+            userCredentials.secretAccessKey = res[3].AccessKey.SecretAccessKey;
+            done();
+        });
+    });
+
+    after(done => {
+        async.series([
+            next => iam.deleteUser({UserName: userName}, next),
+            next => s3.deleteBucket({Bucket: bucketName}, next),
+            next => s3.deleteBucket({Bucket: bucketName2}, next),
+        ], err => {
+            assert.ifError(err);
+            done();
+        });
+    })
+
+    it('should not be able to metadata search without the policy s3:MetadataSearch', done => {
+        return metadataSearchResponseCode(userCredentials, bucketName, (err, res) => {
+            if (err) {
+                assert.ifError(err)
+                return done(err)
+            }
+            assert.strictEqual(res, 403);
```

Comment:
Do you think we can also check if the result is AccessDenied? So we don't only check the http error code (because the bucket ACLs can also lead to 403 responses, maybe?)

Reply:
Also asking because this work is great and will help adding more tests to verify that cloudServer and vault are working well together (e.g. we miss tests with S3+seeded roles and with IAM users)

Reply:
Yeah sure I will take a look if it is possible to be more precise (for sure it is)